### PR TITLE
Fix handling files with non-Ascii characters by sending a TextDocumentIdentifier for dynamic file requests/responses

### PR DESCRIPTION
**Pull request by ryzngard**
VS Code side of https://github.com/dotnet/roslyn/pull/74727

### CODE REVIEW

File: src/lsptoolshost/uriConverter.ts [hunk 1]
```diff
@@ -24,6 +24,6 @@ export class UriConverter {
     }
 
     public static deserialize(value: string): vscode.Uri {
-        return vscode.Uri.parse(value);
+        return vscode.Uri.parse(value, true);
```
Comment (dibarbet) [inline src/lsptoolshost/uriConverter.ts:27]: what does true do here?

Comment (ryzngard) [reply src/lsptoolshost/uriConverter.ts:27]: it's the `strict` parameter. It causes it to throw if the Uri doesn't have a schema

**Review by dibarbet — APPROVED** [review]
pending roslyn update

File: src/razor/src/dynamicFile/removeDynamicFileParams.ts [hunk 2]
```diff
@@ -3,8 +3,9 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { DocumentUri } from 'vscode-languageclient/node';
+import { TextDocumentIdentifier } from 'vscode-languageclient/node';
 
+// matches https://github.com/dotnet/roslyn/blob/9e91ca6590450e66e0041ee3135bbf044ac0687a/src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/RazorDynamicFileInfoProvider.cs#L28
 export class RemoveDynamicFileParams {
-    constructor(public readonly razorFiles: DocumentUri[]) {}
+    constructor(public readonly csharpDocument: TextDocumentIdentifier) {}
```
Comment (ryzngard) [inline src/razor/src/dynamicFile/removeDynamicFileParams.ts:10]: so ironically I named this wrong in the original PR 😅. I'll get that fixed later

**Review by davidwengier — APPROVED** [review]

**Pull request merged**